watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack_conc_recirc_ui.py

```diff
@@ -0,0 +1,634 @@
+# through Lawrence Berkeley National Laboratory, Oak Ridge National Laboratory,
+# National Renewable Energy Laboratory, and National Energy Technology
+# Laboratory (subject to receipt of any required approvals from the U.S. Dept.
+# of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#################################################################################
+from watertap.core.solvers import get_solver
+from watertap.ui.fsapi import FlowsheetInterface
+from watertap.examples.flowsheets.electrodialysis.electrodialysis_1stack_conc_recirc import (
+    build,
+    _condition_base,
+    initialize_system,
+    solve,
+)
+from pyomo.environ import units as pyunits
+
+
+def export_to_ui():
+    return FlowsheetInterface(
+        name="Electrodialysis with concentrate recirculation",
+        do_export=export_variables,
+        do_build=build_flowsheet,
+        do_solve=solve_flowsheet,
+    )
+
+
+def export_variables(flowsheet=None, exports=None, build_options=None, **kwargs):
+    fs = flowsheet
+    # --- Input data ---
+    # Feed conditions
+    exports.add(
+        obj=fs.feed.properties[0].flow_vol_phase["Liq"],
+        name="Feed solution volume flowrate",
+        ui_units=pyunits.m**3 / pyunits.s,
+        display_units="m^3 s^-1",
+        rounding=3,
+        description="Inlet water mass flowrate",
+        is_input=True,
+        input_category="Feed",
+        is_output=False,
+    )
+    exports.add(
+        obj=fs.feed.properties[0].conc_mol_phase_comp["Liq", "Na_+"],
+        name="Na_+ molar concentration",
+        ui_units=pyunits.mol / pyunits.m**3,
+        display_units="mol m^-3",
+        rounding=3,
+        description="Molcar concentration of Na_+ ions in the feed solution",
+        is_input=True,
+        input_category="Feed",
+        is_output=False,
+    )
+
+    exports.add(
+        obj=fs.feed.properties[0].conc_mol_phase_comp["Liq", "Cl_-"],
+        name="Cl_- molar concentration",
+        ui_units=pyunits.mol / pyunits.m**3,
+        display_units="mol m^-3",
+        rounding=3,
+        description="Molcar concentration of Cl_- ions in the feed solution",
+        is_input=True,
+        input_category="Feed",
+        is_output=False,
+    )
+
+    # ED stack conditions
+    exports.add(
+        obj=fs.EDstack.voltage_applied[0],
+        name="Stack voltage",
+        ui_units=pyunits.mol / pyunits.m**3,
+        display_units="V",
+        rounding=3,
+        description="Applied constant voltage on the ED stack",
+        is_input=True,
+        input_category="ED stack",
+        is_output=True,
+    )
+    exports.add(
+        obj=fs.EDstack.cell_pair_num,
+        name="Cell pair number",
+        ui_units=pyunits.dimensionless,
+        display_units="1",
```

Comment:
```suggestion
        display_units="",
```

Reply:
Consider using "1" as the unit of dimensionless qualities as it favors dimensional analysis and is commonly adopted.